src/renderers/noop/ReactNoop.js

```diff
@@ -136,10 +136,22 @@ var NoopRenderer = ReactFiberReconciler({
   },
 
   scheduleAnimationCallback(callback) {
+    if (scheduledAnimationCallback) {
+      throw new Error(
+        'Scheduling an animation callback twice is excessive. ' +
+        'Instead, keep track of whether the callback has already been scheduled.'
+      );
```

Comment:
>Seems unnecessarily dangerous.

Why is it dangerous? It only applies to Noop as a way to keep scheduler stricter.

>If the scheduler calls scheduleAnimationCallback and a callback is already scheduled, the renderer can just treat it as a noop.

It can, but why not enforce scheduler to be explicit instead? Currently ReactDOMFiber would just schedule many idle/animation callbacks if called multiple times. We could fix it there but it seems beneficial to do it in one place (scheduler). This also helps the scheduler because it knows for sure whether it's waiting for something or not, so it can act with more confidence. Also this is already how it works now (it doesn't schedule twice).